### PR TITLE
Upload_values_fix

### PR DESCRIPTION
**Pull request by randrescastaneda**
This PR contains the following: 

1. Fix bug in `update_values()` and `updates_NAs()` where they were not updating multiple variables. 
2. Add different syntax if the input tables are data.table or other. 
3. Update some tests (These two functions are still under-tested) @RossanaTat, could you please add some tests, especially for edge cases? Thanks. 

image

### CODE REVIEW

File: tests/testthat/test-update_values.R [hunk 1]
```diff
@@ -2,44 +2,81 @@
...
+test_that("update_values of one variable", {
 
-  colnames(dt)[3] = "x"
 
   # Updating values in x2$x with values from y2$x
   res <- update_values(dt,
                        var       = "x",
                        reportvar = ".joyn")
+  # check not updated values
+  which(is.na(res$x.x) & res$.joyn != 2) |>
```
Comment (zander-prinsloo) [inline tests/testthat/test-update_values.R]: No values in x should be updated here for dt. Because every row in x either has an NA or the corresponding row in y has an NA.

**Review by zander-prinsloo — APPROVED** [review]
